### ISSUE TITLE
Docs Module not search until open a document

### ISSUE THREAD

**Issue by andmattia** [abpframework/abp#15899]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I have a large repo (more the 1000 md files). I add Elastic search but until I open a document if I click search with a word contained into a document the document is not on search result. So I click and open a document and I found it on search result

### Version

7.0.2

### User Interface

MVC

### Database Provider

EF Core (Default)

### Tiered or separate authentication server

None (Default)

### Operation System

Windows (Default)

**Comment by berkansasmaz** [comment]
As you said, when a document is added, deleted, or updated in the source (GitHub in your case), after a certain time the cache becomes invalidated and the document is automatically re-indexed in Elasticsearch when you visit the relevant document from your documentation website. If you want to see these changes immediately, after updating the document in its source, I recommend that you click on the `Clear cache` and `ReIndex project` buttons, respectively, in the actions of the relevant project from the projects page in the admin section. If you have a lot of documents, I recommend that you do this from the documents page only for the document you have updated.

As a result, if you've updated a document in its source (GitHub in your case) and haven't visited the document on your documentation website for a long time, the new version of the document is not cached and therefore the index in Elasticsearch is outdated. As far as I know, there is no background worker covering this situation. We could solve this case by adding a background worker, but I really don't know if we should handle it. Because we usually tend after updating the document in its source (GitHub in your case), to invalidate the cache and visit the document. Do you agree with me on this?

**Comment by andmattia** [comment]
I try to investigate our use case because I think this worker colud be a good solution. Our case is a docs for internal support and in many case our operators need to search a word in a docs that match the case and in some case this could be involve old docs this actually is out of elastic search scope so it doesn't show on search reasult but if you open the docs (docs move to cache) you can see in result page this confuse some user because search "are not stable" let my say

**Comment by berkansasmaz** [comment]
Yes, I understand that and I agree. Thanks for your suggestion, I'm moving the issue to the backlog 🥳🥳
